[ADD] eCommerce: add product page (#3112)
odoo/documentation: content/applications/websites/ecommerce/managing_products/products.rst

========
Products
========

Odoo allows you to create, import, and manage your products' pages all within the **Website** app.

Add products to the catalog
===========================

To add a product to your catalog, you can either do it in:

- The front end: from the :guilabel:`Homepage`, click :guilabel:`+ New` in the top-right corner, and

>   select :guilabel:`Product`. Enter the name of your product, and :guilabel:`Save`;

Author: Good suggestion!

- The back end: by going to :menuselection:`eCommerce --> Products --> Create`. You can find more
  information on product creation :doc:`here <../../../sales/sales>`;
- By importing: you can :ref:`import data <import-data>` using XLSX or CSV files.

.. seealso::
   - :doc:`../../../sales/sales/products_prices/products/import`

Publish
-------

Upon creation, products are defaulted as :guilabel:`Unpublished` in your eCommerce catalog.
**Unpublished products** are only visible to users with access rights to your eCommerce, whereas
:guilabel:`Published` products are visible to all users. When you wish to render a product visible

> to all users, go to :menuselection:`Site --> Homepage --> Shop`, select the product, and enable the

Author: > by default, we are on the Homepage, isn't it? In this case, I would remove the "Site --> Homepage" or replace "site" by "website app"

You're not on the "Homepage" by default, depending on where you are during the flow. You could be anywhere but the homepage.

> to all users, go to :menuselection:`Site --> Homepage --> Shop`, select the product, and enable the

Author: > by default, we are on the Homepage, isn't it? In this case, I would remove the "Site --> Homepage" or replace "site" by "website app"

By default we are on the homepage, but depending on where in the flow the user is when reading this, I prefer to leave in the full info to avoid ambiguity and misconfusion

:guilabel:`Published` status in the top-right corner.

Product page design
===================

Once a product is created, you can access its **product's page** through :menuselection:`Site -->

> Homepage --> Shop`, selecting the product, and clicking :guilabel:`Edit`. Here, you can change the

Author: I used half the suggestion :)

> **layout** of the page, enable **features**, add **content**, etc. Note that **enabled features**

Reviewer: ```suggestion
**features** of the page, enable **layout**, **add content**, etc. Note that **enabled features**
```
The first paragraph below is related to features so I would mention features first
The section is named "add content" so I would put the "add" in bold too

apply to *all* product pages.

> Features

Reviewer: The structure you propose here is not super clear to me as the Features are also part of the "block" section
![image](https://user-images.githubusercontent.com/85342080/207019649-1c996c15-127b-4146-a4be-b958d42f9663.png)

Author: Good input, I changed the wording

- :guilabel:`Variants`: shows all possible variants of the product as a :guilabel:`Products List`;
  :guilabel:`Options` as selectable options to compose the variant yourself;
- :guilabel:`Cart`: :guilabel:`Buy Now` adds a checkout button taking the customer directly to the
  checkout page; :guilabel:`Wishlist` allows to add the product to a wishlist;
- :guilabel:`Specification`: allows you to select where the **specification** category is displayed.
   This only applies to products with **variants**.

.. note::
   - To allow **wishlists**, the option must be enabled in :menuselection:`Configuration -->
     Settings --> Shop - Products --> Wishlists`;
   - To access the :guilabel:`Variants` options, the :guilabel:`Product Variants` option must first
     be enabled under :menuselection:`Configuration --> Settings --> Shop - Products`. You can learn
     more about variants :doc:`here <../../../sales/sales/products_prices/products/variants>`.

Layout
------

Within the same :guilabel:`Customize` tab as the :ref:`features <ecommerce-features>`, the layout
configuration can be changed according to your needs.

> - :guilabel:`Images Width`: changes the *width* of the image product displayed on the page;

Author: Idk, I must have been high

- :guilabel:`Layout`: the :guilabel:`Carousel` layout displays a large, main image with smaller ones
  underneath; whereas the :guilabel:`Grid` displays four images in a square layout (see pictures
  below);

> - :guilabel:`Image Zoom`: if the zoom occurs on :guilabel:`Pop-up on Click`, when hovering over the

Reviewer: The structure you propose here is not super clear to me as the options mentioned here are not all part of the "customize" section. 
![image](https://user-images.githubusercontent.com/85342080/207019342-11dcc3bc-4962-41cc-89ef-00ca31dc5f23.png)
![image](https://user-images.githubusercontent.com/85342080/207020483-baa82ec0-6f9c-452c-b3d7-f000511c0d83.png)

Author: Not sure where I got those extra options, I tried again and couldn't replicate it

  image (:guilabel:`Magnifier on hover`), on :guilabel:`Both`, or :guilabel:`None`;
- :guilabel:`Image Spacing`: how far apart the images are;
- :guilabel:`Columns`: how many columns of images are displayed;
- :guilabel:`Main Image`: click :guilabel:`Replace` to change the product's main image;
- :guilabel:`Extra Images`: click :guilabel:`Add` or :guilabel:`Remove all` to add or remove product
  images. You can also add images and videos via URL.

.. note::
   Images must be either PNG or JPG and at least 1042px in height or width. To trigger the zoom, the
   image has to be bigger than 1024x1024

.. image:: products/products-layout.png
   :align: center
   :alt: Product images layout

Add content
-----------

> You can use **building blocks** to add content to your product page. These blocks can be used to add

Reviewer: Where to select building blocks? Is there a "building blocks" option available somewhere? should we enable this feature? Could you please clarify? :)

extra text and picture galleries, features such as call to actions or comparisons, etc. Depending on
*where* you drop the **building block**, it may be available either on the *product page* only, or
on the *whole* website.

.. image:: products/products-blocks.png
   :align: center
   :alt: Building blocks on product page

Product configuration
=====================

Multiple languages
------------------

If multiple languages are available on your website and you wish to have the product's information
translated, it is necessary to encode these translated information in the **product's form**. Fields
with multiple languages available are identifiable by their abbreviation language (ex. :abbr:`EN

> (English)`) next to their field.

Author: It is because (as written) found in the **product's form** and not the **product's page**. I changed the wording so it is less confusing. Ideally, there'd be a :ref: to explain how to check that, but the page actually doesn't exist. I'll add or a note or something in the meantime

.. image:: products/products-field-translation.png
   :align: center
   :alt: Field translation

The **eCommerce-related** fields to translate are:

- :guilabel:`Product name`;
- :guilabel:`Out-of-Stock Message` (under the :guilabel:`Sales` tab);
- :guilabel:`Sales Description` (under the :guilabel:`Sales` tab);

.. tip::
   Having untranslated content on a web page may be detrimental to the user experience and
   therefore, your SEO.

Website availability
--------------------

A product can be available on *one* or *all* websites, but it is not possible to select *some*
websites and not others. To define a product's availability, go to
:menuselection:`eCommerce --> Products`, select your product, and in the :guilabel:`Sales` tab,
click the :guilabel:`Website` you wish the product to be available on. Leave it empty for *all*.

Digital files
-------------

> It is possible to add a download link for customers at the end of the checkout. To do that, first

Reviewer: what's the purpose? add an example?

========
Products
========

Odoo allows you to create, import, and manage your products' pages all within the **Website** app.

Add products to the catalog
===========================

To add a product to your catalog, you can either do it in:

- :menuselection:`Website app --> Homepage`, click :guilabel:`+ New` in the top-right corner, and
  select :guilabel:`Product`. Enter the name of your product, and :guilabel:`Save`;
- :menuselection:`Website app --> eCommerce --> Products --> Create`, you can find more information

>   on product creation :doc:`here <../../../sales/sales>`;

Author: Done

- by :ref:`importing data <import-data>` using XLSX or CSV files.

.. seealso::
   - :doc:`../../../sales/sales/products_prices/products/import`

Publish
-------

Upon creation, products are defaulted as :guilabel:`Unpublished` in your eCommerce catalog.
**Unpublished products** are only visible to users with access rights to your eCommerce, whereas
:guilabel:`Published` products are visible to all users. When you wish to render a product as
visible to all users, go to :menuselection:`Website --> Site --> Homepage --> Shop`, select the
product, and enable it as :guilabel:`Published` in the top-right corner.

Product page design
===================

Once a product is created, you can access its **product page** through the :guilabel:`Shop` page by
clicking on the product, and then clicking :guilabel:`Edit`. Here, you can change the **additional
functions** of the page, change the **layout**, **add content**, etc. Note that **enabled
functions** apply to *all* product pages.

Additional functions
--------------------

.. _ecommerce-functions:

In the **website builder** window, click :guilabel:`Customize` to enable additional functions such

> as :guilabel:`Rating`, :guilabel:`Share`, etc.

Author: Good suggestion

  underneath; whereas the :guilabel:`Grid` displays four images in a square layout (see pictures
  below);
- :guilabel:`Image Zoom`: if the zoom occurs on :guilabel:`Pop-up on Click`, when hovering over the
  image (:guilabel:`Magnifier on hover`), on :guilabel:`Both`, or :guilabel:`None`;
- :guilabel:`Thumbnails`: if you wish to align the thumbnails **vertically** (:guilabel:`Left`), or
  **horizontally** (:guilabel:`Right`);
- :guilabel:`Main Image`: click :guilabel:`Replace` to change the product's main image;
- :guilabel:`Extra Images`: click :guilabel:`Add` or :guilabel:`Remove all` to add or remove product
  images. You can also add images and videos via **URL**.

.. note::
   Images must either be PNG or JPG, and at least 1042px in height or width. To trigger the zoom,
   the image has to be bigger than 1024x1024.

.. image:: products/products-layout.png
   :align: center
   :alt: Product images layout

Add content
-----------

You can use **building blocks** (:menuselection:`Edit --> Blocks`) to add content to your product
page. These blocks can be used to add extra text and picture galleries, features such as

> :guilabel:`Call to Actions`, **comparisons**, etc. Depending on *where* you drop the **building

Reviewer: ```suggestion
:guilabel:`Call to Actions`, :guilabel:`Comparisons`, etc. Depending on *where* you drop the **building
```

block**, it may be available either on the product page *only*, or on the *whole* website.

.. image:: products/products-blocks.png
   :align: center
   :alt: Building blocks on product page

Product configuration
=====================

Multiple languages
------------------

If multiple languages are available on your website and you wish to have the product's information

> translated, it is necessary to encode these translated information in the **product's template**.

Reviewer: ```suggestion
translated, it is necessary to encode this translated information in the **product's template**.
```

Reviewer: "this" as information is singular

Fields with multiple languages available are identifiable by their abbreviation language (ex.
:abbr:`EN (English)`) next to their field.

.. image:: products/products-field-translation.png
   :align: center
   :alt: Field translation

The **eCommerce-related** fields to translate are:

- :guilabel:`Product name`;
- :guilabel:`Out-of-Stock Message` (under the :guilabel:`Sales` tab);
- :guilabel:`Sales Description` (under the :guilabel:`Sales` tab);

.. tip::
   Having untranslated content on a web page may be detrimental to the user experience and

>    therefore, your SEO.

Reviewer: Is it really a tip?

Author: What else could it be ? A note?

.. note::
   To check the language(s) of your website, go to :menuselection:`Website app --> Configuration -->

>    Settings --> Website Info`.

Reviewer: ```suggestion
   Settings --> Website Info section`.
```

.. seealso::
   - :ref:`Multi-language support <seo-multilanguage>`

Website availability
--------------------

A product can be set available on either *one* or *all* websites, but it is not possible to select

> *some* websites and not others. To define a product's availability, go to
> :menuselection:`Website app --> eCommerce --> Products`, select your product, and in the
> :guilabel:`Sales` tab, click the :guilabel:`Website` you wish the product to be available on. Leave
> it empty for *all*.

Reviewer: ```suggestion
*some* websites and not others. To define a product's availability, go to
:menuselection:`Website app --> eCommerce --> Products`, select your product, and in the
:guilabel:`Sales` tab, click the :guilabel:`Website` you wish the product to be available on. Or leave
this field empty if you want to set your product available on *all* websites.
```

Files` and :guilabel:`Create` a new file.

.. image:: products/products-digital-files.png
   :align: center
   :alt: Digital Files menu

For the configuration:

- :guilabel:`Name`: the name of your file;
- :guilabel:`Type:` select if it is either a **file** or a **URL**. Accordingly, you either have a
  :guilabel:`File Content (base64)` field to upload your file, or a :guilabel:`URL` field to enter
  your URL.
- :guilabel:`Website`: The website on which that file is *available*. If you want it available for
  *all* websites, leave it empty.

The file is then available in the **purchase order** on the customer's portal, after the checkout.

Stock management
================

Under the :menuselection:`Website app --> Configuration --> Settings --> Shop - Products`, you can
enable and configure inventory management options.

.. important::
   To display the stock level on the product page, the :guilabel:`Product Type` on the **product's
   form** must be set to :guilabel:`Storable` (only available when the **Inventory** app is
   installed).

In the :guilabel:`Inventory Defaults` sub-section, you can select the eCommerce selling strategy of
products:

> - :guilabel:`Out-of-stock (Continue Selling)`: enabling it allows customers to continue placing

Reviewer: ```suggestion
- :guilabel:`Out-of-Stock (Continue Selling)`: enabling it allows customers to continue placing
```

  orders even when the produc is **out-of-stock**. Leave it  unchecked to prevent orders;
- :guilabel:`Show Available Qty`: enabling it displays the available quantity left under a specified
  threshold on the product page. The available quantity is calculated based on the "On hand"
  quantity minus the quantity already reserved for outgoing transfers.

Additionally, you can **prevent the sale** of a product if its price equals `0`. To do so, go to
:menuselection:`Website app --> Configuration --> Settings --> Shop - Products`, and enable
:guilabel:`Prevent Sale of Zero Priced Product`. This replaces the :guilabel:`Add to Cart` button by
a :guilabel:`Contact us` button.

Selling as kit
--------------

> If you are selling non-prepackaged kits (i.e, the kits are made of individual products), we

Reviewer: ```suggestion
If you are selling non-prepackaged kits (i.e., the kits are made of individual products), we
```

recommend you read :doc:`this documentation
<../../../inventory_and_mrp/manufacturing/management/kit_shipping>` to keep track of your stock.

Product comparison
==================

You can enable a **product comparison tool** for your eCommerce by going to
:menuselection:`Website app --> Configuration --> Settings --> Shop - Products`, and ticking
:guilabel:`Product Comparison Tool`. This tool allows to save products' **specifications** and
compare them against each other on a single page.

> On the product page, scroll down to the :guilabel:`Specifications` section, and click

Reviewer: ```suggestion
On the product page, scroll down to the :guilabel:`Specifications` section and click
```

========

> Products

Author: Good suggestion, I swapped the names around. I'll see for the other names once the mega PR is to be pushed.

:guilabel:`Call to Actions`, :guilabel:`Comparisons`, etc. Depending on *where* you drop the
**building block**, it may be available either on the product page *only*, or on the *whole*
website.

.. image:: products/products-blocks.png
   :align: center
   :alt: Building blocks on product page

Download link
-------------

If you wish to add a downloadable file (ex.: user's manual, notice of use, etc.) on the product
page, you can do so by adding a :guilabel:`Text` block from :menuselection:`Edit --> Blocks` on the
page. Once placed, click within the :guilabel:`Text` block, and under the :guilabel:`Inline Text`
section, select either :menuselection:`Insert Media --> Documents` or :guilabel:`Insert or edit
link` and enter the URL in the :guilabel:`Your URL` field.

.. note::
   The difference with :ref:`digital files <ecommerce-digital-file>` is that digital files can only
   be downloaded *after* checkout.

.. image:: products/products-media.png
   :align: center
   :alt: Media and link buttons

.. important::
   The :ref:`developer mode <developer-mode>` is intended only for **advanced users** who want to

>    have access to advanced and optional features.

Author: If I recall correctly, Jon wants that users know the "developer mode" is not the intended working way and generally shouldn't even be in the doc. However, some options are just stuck behind the "dev mode."  For a user that reads the complete doc, it may get repetitive, but for someone specifically looking for a piece of information and landing on this page, I believe the "dev" note should be indicated.

I'm changing it from "Important" to "Note".

P.S.: Since it was linked to the studio "trick" that was removed, I removed it too.

Alternatively, you can use the :ref:`developer mode <developer-mode>` and the **Studio** app to add
an extra field on the **product template** of the product. On the **product template** of the
product you wish to add images on, click on :menuselection:`Toggle Studio --> Sales tab` and from

> the :guilabel:`+ Add`, drag a :guilabel:`File` field to the :guilabel:`eCommerce shop` section.

Author: It was an example discussed in the smartclass, but I also believe it is very niche. I'm removing it.

Stock management
================

Under the :menuselection:`Website --> Configuration --> Settings --> Shop - Products`, you can
enable and configure inventory management options.

.. important::
   To display the stock level on the product page, the :guilabel:`Product Type` on the **product's
   form** must be set to :guilabel:`Storable` (only available when the **Inventory** app is
   installed).

Inventory
---------

In the :guilabel:`Inventory Defaults` sub-section, you can select the eCommerce selling strategy of
products:

- :guilabel:`Warehouse`: if you have multiple warehouses, you can define the warehouse associated to
  your website. If you have multiple websites, you can select a different one per website;
- :guilabel:`Out-of-Stock (Continue Selling)`: enabling it allows customers to continue placing
  orders even when the product is **out-of-stock**. Leave it  unchecked to **prevent orders**;
- :guilabel:`Show Available Qty`: enabling it displays the available quantity left under a specified
  threshold on the product page. The available quantity is calculated based on the "On hand"
  quantity minus the quantity already reserved for outgoing transfers.

> Additionally, you can **prevent the sale** of a product if its price equals `0`. To do so, go to

Author: Added as to do, since the doc isn't in the same branch and isn't published yet

:menuselection:`Website --> Configuration --> Settings --> Shop - Products`, and enable
:guilabel:`Prevent Sale of Zero Priced Product`. This replaces the :guilabel:`Add to Cart` button by
a :guilabel:`Contact us` button.

Selling as kit
--------------

If you are selling non-prepackaged kits (i.e., the kits are made of individual products), we
recommend you read the related documentation to keep track of your stock.

.. seealso::
   :doc:`../../../inventory_and_mrp/manufacturing/management/kit_shipping`

Product comparison
==================

You can enable a **product comparison tool** for your eCommerce by going to
:menuselection:`Website --> Configuration --> Settings --> Shop - Products`, and ticking
:guilabel:`Product Comparison Tool`. This tool allows to save products' **specifications** and
compare them against each other on a single page.

On the product page, scroll down to the :guilabel:`Specifications` section and click
:guilabel:`Compare`. Repeat the same process for all products you wish to compare. Then, click the
:guilabel:`Compare` button of the pop-up window at the bottom of the page to reach the comparison
summary.

.. note::

>    The :guilabel:`Product Comparison Tool` works based on **attributes**, and therefore can only be
>    used with products which have **attributes** specified in their **product's template**.

Reviewer 2: ```suggestion
   used with products which have **attributes** specified in their **product's template**.
```
```suggestion
   The :guilabel:`Product Comparison Tool` can only be used if **attributes** are set on the **product's template**.
```
Could be simplified

  :doc:`variants <../../../sales/sales/products_prices/products/variants>` of the product as a
  :guilabel:`Products List`;
  :guilabel:`Options` as selectable options to compose the variant yourself;
- :guilabel:`Cart`: :guilabel:`Buy Now` adds a checkout button taking the customer directly to the
  checkout page; :guilabel:`Wishlist` allows to add the product to a wishlist;
- :guilabel:`Specification`: allows you to select where the :guilabel:`Specifications` section is
  displayed. This option displays a list of all variant attributes and values of a product, but only
  works for products *with* variants.

.. note::
   - To allow **wishlists**, the option must be enabled in :menuselection:`Website --> Configuration
     --> Settings --> Shop - Products`;
   - To access the :guilabel:`Variants` options, the :doc:`Product Variants
     <../../../sales/sales/products_prices/products/variants>` option must first be enabled under
     :menuselection:`Website --> Configuration --> Settings --> Shop - Products`.

Layout
------

Within the same :guilabel:`Customize` tab as the :ref:`functions <ecommerce-functions>`, the layout
configuration can be changed according to your needs.

> - :guilabel:`Images Width`: changes the width of the images product displayed on the page;

Reviewer 2: ```suggestion
- :guilabel:`Images Width`: changes the width of the product images displayed on the page;
```

- :guilabel:`Type:` select if it is either a **file** or a **URL**. Accordingly, you either have a
  :guilabel:`File Content (base64)` field to upload your file, or a :guilabel:`URL` field to enter
  your URL.
- :guilabel:`Website`: the website on which that file is *available*. If you want it available for
  *all* websites, leave it empty.

The file is then available after checkout in the **purchase order** section found on the customer's
portal.

Product configuration
=====================

Multiple languages
------------------

If multiple languages are available on your website and you wish to have the product's information
translated, it is necessary to encode this translated information in the **product's template**.
Fields with multiple languages available are identifiable by their abbreviation language (ex.
:abbr:`EN (English)`) next to their field.

.. image:: products/products-field-translation.png
   :align: center
   :alt: Field translation

The **eCommerce-related** fields to translate are:

- :guilabel:`Product name`;
- :guilabel:`Out-of-Stock Message` (under the :guilabel:`Sales` tab);
- :guilabel:`Sales Description` (under the :guilabel:`Sales` tab);

> .. note::
>    Having untranslated content on a web page may be detrimental to the user experience and
>    :doc:`SEO <../../../websites/website/optimize/seo>`.
> 
> .. note::
>    To check the language(s) of your website, go to :menuselection:`Website --> Configuration -->
>    Settings --> Website Info section`.

Reviewer 3: ```suggestion
.. note::
   - Having untranslated content on a web page may be detrimental to the user experience and
   :doc:`SEO <../../../websites/website/optimize/seo>`.
   - To check the language(s) of your website, go to :menuselection:`Website --> Configuration -->
   Settings --> Website Info section`.
```
